scripts/tests/twister/test_config_parser.py

```diff
@@ -0,0 +1,265 @@
+    common:
+      filter: 'filter2'
+    '''
+
+    loaded_schema = scl.yaml_load(
+        os.path.join(zephyr_base, 'scripts', 'schemas','twister', 'testsuite-schema.yaml')
+    )
+
+    with mock.patch('builtins.open', mock.mock_open(read_data=yaml_data)):
+        parser = TwisterConfigParser(filename, loaded_schema)
+        parser.load()
+
+    scenario_data = parser.get_scenario('scenario1')
+    scenario_common = parser.common
+
+    assert scenario_data['tags'] == {'tag1', 'tag2'}
+    assert scenario_data['extra_args'] == ['--CONF_FILE=file1.conf', '--OVERLAY_CONFIG=config1.conf']
+    assert scenario_common == {'filter': 'filter2'}
+
+
+def test_default_values(zephyr_base):
+    filename = "test_data.yaml"
+
+    yaml_data = '''
+    tests:
+      scenario1:
+        tags: 'tag1'
+        extra_args: ''
+    '''
+
+    loaded_schema = scl.yaml_load(
+        os.path.join(zephyr_base, 'scripts', 'schemas', 'twister','testsuite-schema.yaml')
+    )
+
+    with mock.patch('builtins.open', mock.mock_open(read_data=yaml_data)):
+        parser = TwisterConfigParser(filename, loaded_schema)
+        parser.load()
+
+    scenario_data = parser.get_scenario("scenario1")
+    assert scenario_data["type"] == "integration"
+    assert scenario_data["extra_args"] == []
+    assert scenario_data["extra_configs"] == []
+    assert scenario_data["extra_conf_files"] == []
+    assert scenario_data["extra_overlay_confs"] == []
+    assert scenario_data["extra_dtc_overlay_files"] == []
+    assert scenario_data["required_snippets"] == []
+    assert scenario_data["build_only"] is False
+    assert scenario_data["build_on_all"] is False
+    assert scenario_data["skip"] is False
+    assert scenario_data["slow"] is False
+    assert scenario_data["timeout"] == 60
+    assert scenario_data["min_ram"] == 8
+    assert scenario_data["modules"] == []
+    assert scenario_data["depends_on"] == set()
+    assert scenario_data["min_flash"] == 32
+    assert scenario_data["arch_allow"] == set()
+    assert scenario_data["arch_exclude"] == set()
+    assert scenario_data["extra_sections"] == []
+    assert scenario_data["integration_platforms"] == []
+    assert scenario_data["ignore_faults"] is False
+    assert scenario_data["ignore_qemu_crash"] is False
+    assert scenario_data["testcases"] == []
+    assert scenario_data["platform_type"] == []
+    assert scenario_data["platform_exclude"] == set()
+    assert scenario_data["platform_allow"] == set()
+    assert scenario_data["platform_key"] == []
+    assert scenario_data["toolchain_exclude"] == set()
+    assert scenario_data["toolchain_allow"] == set()
+    assert scenario_data["filter"] == ""
+    assert scenario_data["levels"] == []
+    assert scenario_data["harness"] == "test"
+    assert scenario_data["harness_config"] == {}
+    assert scenario_data["seed"] == 0
+    assert scenario_data["sysbuild"] is False
+
+
+def test_cast_string_to_string(zephyr_base):
+    value = "hello"
```

Comment:
You forgo testing the `.strip()` path ([L104 in config_parser.py](https://github.com/zephyrproject-rtos/zephyr/blob/aa87ed5d8a711c50c5db325b543068ac583d2a6b/scripts/pylib/twister/twisterlib/config_parser.py#L104-L105)) with such a value. I'd add some spaces.

Reply:
good point!